solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/stream/CommitStream.java

```diff
@@ -55,6 +55,7 @@ public class CommitStream extends TupleStream implements Expressible {
   private TupleStream tupleSource;
 
   private transient SolrClientCache clientCache;
+  private transient boolean isCloseCache;
```

Comment:
Do we need this new boolean `isCloseCache`? Can't we just check if `clientCache == null` in the close method and set it to `clientCache = null` after `clientCache.close();`?

Reply:
you are right. there are 2 patterns of doing this in the PR (flag vs null check). I will switch to this one, it's cleaner.

Reply:
sorry @risdenk I misread your question. in this case no, you can't get rid of the flag because if the cache is not null you don't know if it came from the context or it was created locally. if it came from the context it's reused for multiple streams so you can't close it here, you have to close it at the parent location where it was created.

Reply:
The name "isCloseCache" is clumsy IMO; I think "doCloseCache" is far clearer.